SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/WorkerCoordinator/LifetimeComponent.cs

```diff
@@ -0,0 +1,208 @@
+        public long EndTick;
+    }
+
+    internal interface ILifetimeComponentHost
+    {
+        void StartClient(ClientInfo clientInfo);
+        void StopClient(ClientInfo clientInfo);
+
+        void CheckPlayerStatus();
+    }
+
+    internal class LifetimeComponent
+    {
+        // Arguments for lifetime management.
+        public const string MaxLifetimeArg = "max_lifetime";
+        public const string MinLifetimeArg = "min_lifetime";
+        public const string RestartAfterSecondsArg = "restart_after_seconds";
+        public const string UseNewSimulatedPlayerArg = "use_new_simulated_player";
+
+        // Lifetime management parameters.
+        private ILifetimeComponentHost Host;
+        private bool UseNewSimulatedPlayer;
+        private int MaxLifetime;
+        private int MinLifetime;
+        private int RestartAfterSeconds;
+        private List<ClientInfo> WaitingList;
+        private List<ClientInfo> RunningList;
+
+        private int TickIntervalSeconds = 1;
+        private Logger Logger;
+        private Random Random;
+
+        // Temp variables to avoid allocate variable in tick method.
+        private long CurTicks;
+        private int Length;
+        private ClientInfo SimulatedClientInfo;
+
+        /// <summary>
+        /// Create lifetime component, will return null if max lifetime argument not in args.
+        /// </summary>
+        /// <param name="logger"></param>
+        /// <param name="args"></param>
+        /// <param name="numArgs"></param>
+        /// <returns></returns>
+        public static LifetimeComponent Create(Logger logger, string[] args, out int numArgs)
+        {
+            numArgs = 0;
+
+            // Max lifetime argument.
+            int maxLifetime = 0;
+            if (Util.HasIntegerArgument(args, MaxLifetimeArg))
+            {
+                maxLifetime = Util.GetIntegerArgument(args, MaxLifetimeArg);
+                numArgs++;
+            }
+
+            // Use max lifetime as default.
+            int minLifetime = maxLifetime;
+            if (Util.HasIntegerArgument(args, MinLifetimeArg))
+            {
+                minLifetime = Util.GetIntegerArgument(args, MinLifetimeArg);
+                numArgs++;
+            }
+
+            // Use 60 seconds as default.
+            int restartAfterSeconds = 60;
+            if (Util.HasIntegerArgument(args, RestartAfterSecondsArg))
+            {
+                restartAfterSeconds = Util.GetIntegerArgument(args, RestartAfterSecondsArg);
+                numArgs++;
+            }
+
+            // Default do not use new simulated player to restart.
+            int useNewSimulatedPlayer = 0;
+            if (Util.HasIntegerArgument(args, UseNewSimulatedPlayerArg))
+            {
+                useNewSimulatedPlayer = Util.GetIntegerArgument(args, UseNewSimulatedPlayerArg);
+                numArgs++;
+            }
+
+            // Disable function by do not define max lifetime.
```

Comment:
It seems OK to pass 0 and always rely on the LifetimeComponent. This way, we won't need to support two separate systems.

Reply:
This feature is an option. We don't want to affect the configuration that is currently in use. The default value of `max_lifetime` is 0.  Not setting `max_lifetime` allows us to not have to modify those configuration files that are already in existence.

Reply:
Sure - this is why I'd handle `max_lifetime == 0` inside of the `LifetimeComponent` to never kill simplayers, instead of handling it externally and having two different execution paths with and without the `LifetimeComponent`.